File: Cpp Codes/Find_theFine.cpp
#include<bits/stdc++.h>
using namespace std;

int CollectFine(int *crr, int *frr, int n, int date)
{
    int i, fine=0; 
    if(date % 2 == 0)
    {
        for(i=0; i<n; i++)
        {
            if(crr[i] % 2 != 0)
                fine = fine+frr[i];
        }
    }
    else
    {
        for(i=0; i<n; i++)
        {
            if(crr[i] % 2 == 0)
                fine = fine+frr[i];
        }
    }
    return fine;
}


int main() 
{
	int t;

    cout<<"\nEnter the testcases: ";
	cin>>t;
	
	while(t)
	{
	    int n, date, i;
	    cout<<"\nEnter the Size and Date: ";
        cin>>n>>date;
	    
	    int crr[n], frr[n];
        cout<<"Enter the Cars: ";
	    for(i=0; i<n; i++)
	        cin>>crr[i];
	        
        cout<<"Enter the Fine: ";
        for(i=0; i<n; i++)
	        cin>>frr[i];
        
        int fine = CollectFine(crr, frr, n, date);
        cout<<"Total Fine is: "<<fine<<"\n";
        
        t--;
	}
	
	return 0;
}

File: Array/Find_the_Fine.py
# Given an array of penalties P, an array of car numbers C, and also the date D.
# The task is to find the total fine which will be collected on the given date.
# Fine is collected from odd-numbered cars on even dates and vice versa.

def CollectFine(Crr, Frr, n, date):
    if date % 2 == 0:
        fine = 0
        for i in range(n):
            if Crr[i] % 2 != 0:
                fine = fine+Frr[i]
    else:
        fine = 0
        for i in range(n):
            if Crr[i] % 2 == 0:
                fine = fine+Frr[i]
    

    return fine


def FineCollection(Crr, Frr, date):
    fine = 0
    if date % 2 == 0:
        i = 0
        while i < len(Crr):
            if Crr[i] % 2 != 0:
                fine = fine + Frr[i]
            i += 1
    else:
        j = 0
        while j < len(Crr):
            if Crr[j] % 2 == 0:
                fine = fine + Frr[j]
            j += 1
    
    return fine


if __name__ == "__main__":
    print("\n")

    # date = 12
    # Crr = [2375, 7682, 2325, 2352]
    # Frr = [250, 500, 350, 200]
    # n = len(Crr)

    date = 9
    Crr = [2222,2223,2224]
    Frr = [200,300,400]
    n = len(Crr)

    # fine = CollectFine(Crr, Frr, n, date)
    fine = FineCollection(Crr, Frr, date)
    print(fine)

    print("\n")
